mesonbuild/interpreter/interpreter.py

```diff
@@ -1430,6 +1431,21 @@ def compatibility_sort_helper(s):
         if main_summary:
             main_summary.dump()
 
+    @typed_pos_args('hash', varargs=str, min_varargs=2, max_varargs=2)
+    @noKwargs
+    def func_hash(self, node: mparser.BaseNode, args: T.Tuple[T.List[str]], kwargs: 'TYPE_kwargs') -> str:
+        from hashlib import new
+        string, hash_type = args[0][0:2]
+        # For supported algorithms
+        # see https://docs.python.org/3/library/hashlib.html#hash-objects
+        if hash_type not in ['md5', 'sha1',
```

Comment:
Just catch the ValueError from `hashlib.new` and rethrow its message as an InterpreterException (`raise InterpreterException(str(e))` should work).

Also add a test to `test cases/common/35 string operations`, including one for the error

```
testcase expect_error(...)
  'abc'.hash('invalid-method')
endtestcase
```